### PR TITLE
Improvements for week 6

### PR DESCRIPTION
**Pull request by dpshelio**
There's been a couple of things that were failing
- git bisect because the test repository is not being cloned
- the js and css of the built documentation not being shown.

This PR attempts to fix these problems and adds some minor updates to how some information is being displayed (automated docs generation and better signposting of voluntary introduced errors).

### CODE REVIEW

File: ch04packaging/04documentation.ipynb [hunk 1]
```diff
@@ -61,15 +61,22 @@
...
+    "| Fortan   | [FORD](https://github.com/Fortran-FOSS-Programmers/ford)                            | [`arange` docs](https://stdlib.fortran-lang.org/interface/arange.html)              | [`arange` docstring source](https://github.com/fortran-lang/stdlib/blob/d14fca8e7cc36ed5f6f84d2bf576c91c2e54eb07/src/stdlib_math.fypp#L276-L290) |\n",
+    "| Rust     | [rustdoc](https://doc.rust-lang.org/rustdoc/what-is-rustdoc.html)                   | [`Matrix` docs](https://docs.rs/nalgebra/0.18.0/nalgebra/base/struct.Matrix.html)   | [`Matrix` docstring source](https://github.com/dimforge/nalgebra/blob/8ea8ac70d5ad4bae865e6246a48455bf0b3fa3d2/src/base/matrix.rs#L59-L157)      |\n",
+    "\n",
+    "[Breathe](https://breathe.readthedocs.io/en/latest/) can be used to make Sphinx and Doxygen work together (good to keep documentation, for example, of a c++ project that includes python bindings). [roxygen2](https://cran.r-project.org/web/packages/roxygen2/vignettes/roxygen2.html) is another good option for R packages.\n",
```
Comment (ageorgou) [inline ch04packaging/04documentation.ipynb:77]:
```suggestion
    "[Breathe](https://breathe.readthedocs.io/en/latest/) can be used to make Sphinx and Doxygen work together (good to keep documentation, for example, of a C++ project that includes Python bindings). [roxygen2](https://cran.r-project.org/web/packages/roxygen2/vignettes/roxygen2.html) is another good option for R packages.\n",
```

File: ch04packaging/04documentation.ipynb [hunk 2]
```diff
@@ -409,7 +446,7 @@
    "source": [
     "`doctest` is a module included in the standard library. It runs all the code within the docstrings and checks whether the output is what it's claimed on the documentation.\n",
     "\n",
-    "Let's add an example to our greeting function and check it with `doctest`. We are leaving the output with a small typo to see what's the type of output we get from `doctest`."
+    "Let's add an example to our greeting function and check it with `doctest`. We are leaving the output with a small typo (missing the close quote `'`) to see what's the type of output we get from `doctest`."
```
Comment (ageorgou) [inline ch04packaging/04documentation.ipynb:449]:
```suggestion
    "Let's add an example to our greeting function and check it with `doctest`. We are leaving the output with a small typo (missing the closing quote `'`) to see what's the type of output we get from `doctest`."
```

File: .gitignore [hunk 3]
```diff
@@ -2,10 +2,15 @@
...
 _site/
+build/
+workflow/
+vendor/
 .sconsign.dblite
 html/index.html
 pdf
 *.pyc
+*.nbconvert.ipynb
+*.html
```
Comment (ageorgou) [inline .gitignore:13]: Is this going to cause any problems for the `gh-pages` branch, or are the generated pages force-added?

**Review by ageorgou — APPROVED** [review]
Looks good, only question is for the `.gitignore`.

File: .gitignore (hunk 3 above)
Comment (dpshelio) [reply .gitignore:13]: Right! maybe the `html` shouldn't be there!

Comment (dpshelio) [inline .gitignore:13]:
```suggestion
```

**Pull request merged**